### PR TITLE
Impl. of MPL LinearProperties.

### PR DESCRIPTION
**Pull request by TomFischer**

### CODE REVIEW

**Comment by TomFischer** [comment]
@renchao-lu Can you review the implementation, please?

**Comment by renchao-lu** [comment]
Looks fine to me. @TomFischer

File: MaterialLib/MPL/Properties/LinearProperty.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+struct IndependentVariable
+{
+    Variables type;
+    VariableType reference_condition; // scalar or vector
+    VariableType slope; // scalar or matrix
+};
+
+/// The linear property class. This property calculates the linear relationship.
+/// constant value. It accepts all datatypes defined in PropertyDataType
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply MaterialLib/MPL/Properties/LinearProperty.h]: Fixed.

File: MaterialLib/MPL/Properties/LinearProperty.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+/// (currently: double, Vector, Tensor, std::string)
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply MaterialLib/MPL/Properties/LinearProperty.h]: Corrected documentation.

File: Tests/MaterialLib/TestMPLLinearProperty.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    MaterialPropertyLib::VariableArray variable_array;
+    variable_array[MaterialPropertyLib::Variables::temperature] = 303.15;
+    ASSERT_NEAR(
+        boost::get<double>(linear_property.value(variable_array)),
+        y_ref + m * (boost::get<double>(
+                         variable_array
+                             [MaterialPropertyLib::Variables::temperature]) -
+                     x_ref),
+        1.e-10);
+    ASSERT_NEAR(
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply Tests/MaterialLib/TestMPLLinearProperty.cpp]: Use now `ASSERT_EQ`.

**Comment by TomFischer** [comment]
@wenqing  and @endJunction for review. I'll rebase now and merge if all tests are green.

**Pull request merged**